### DISCUSSION TITLE
Disable "had recent pushes" banner for some branches

### DISCUSSION THREAD

**Discussion by metaturso** [community/community#35200] [Repositories]
When protecting branches it should be possible to specify a branch to never be treated as a source (_compare_) side of a Pull Request.

This is a common scenario where `develop` is merged into `master` or `main`, at the end of a change, to trigger specific CD release workflows, and changes are never merged from `master` into `develop`.

Declaring the `develop` branch instead of `master` as the **default** branch is useful. GitHub shows the content of this branch on the repository page, and it prevents new Pull Request from accidentally targeting the `master` branch.

The problem with this approach is this banner that makes it easy to think `master` needs merging back into `develop`. The only difference between the two branches is the merge commit generated by the pull request.

![github_ignore_merges_to_master](https://user-images.githubusercontent.com/30650/194085579-0bd2f309-81e5-4df4-bbfb-5ab617b22c6c.PNG)

Please make it possible to disable this banner for selected branches.

> **Note**
> This request differs from #34225, which asks for the banner to be completely disabled. I believe this is a useful banner. Just not for all branches.

**Comment by OwensCode** [comment]
I vote for this change

**Reply by inzayn99** [reply]
Response #1

**Reply by inzayn99** [reply]
Response #2

**Comment by jrett** [comment]
I agree with this.
But can we simply not recommend a pull request if there are no actual changes between the branches?
I keep getting this banner, and when I look at the other branch, I get it saying that one or the other is 1 commit behind or ahead of the other, and it suggests I do a pull request. But then at the bottom of the screen, it shows no differences.

Or when I PR and merge into one branch, between the two they still say there's a difference, and when it shows the difference, it's comparing with a previous commit. Am I missing something?

**Reply by olafur-palsson** [reply]
I had this also, in my case one of them had a merge commit and the other not. It's able to merge but will not do much.

**Reply by dgchrt** [reply]
This would sort both issues. If `staging` and `production` are always behind `main` anyways, there should be no automatic banner for them.

**Comment by SubvertDev** [comment]
That would be awesome, kinda annoying when your main branch is develop and you push something to "main"

**Reply by karol767** [reply]
Yes, I'm interested in Dev Branch Composing.
Ask for your opinion.

**Comment by Jorann** [comment]
Just ran into this suggestion. I was surprised this is not an option in branch protection rules.

**Reply by lukemorton** [reply]
Yep same here!

**Comment by c-reeder** [comment]
Has there been any movement on this? I also think this would be a great feature especially for branches that have automatic processes occurring on them.

**Comment by bwees** [comment]
I have a GitHub action that updates data in a separate branch and its a pain to see this every time I log in.

**Reply by carry0987** [reply]
Same here, I don't want to see `gh-pages had recent pushes` anymore

**Comment by NilsBaumgartner1994** [comment]
+1 would like to see that too.

**Comment by dontic** [comment]
I agree this is a pain for the common "develop" > "staging" > "master" branch workflow where the active branch is the "develop" one. This situation is quite common.

**Reply by dgchrt** [reply]
No need to go that far with customization. It also happens for the vanilla `main` -> `staging` -> `production` workflow.

**Comment by byronbowerman** [comment]
This is also unhelpful for orphan branches that shouldn't merge back into the default/main/master/trunk

**Reply by bwees** [reply]
Yeah I have code that is completely separate from the main branch that does something completely different that I still want in the same repo but not alongside the main code

**Comment by bsconnor** [comment]
+1 would be great to see

**Comment by eduardoabpereira** [comment]
+1

**Comment by shreyass-ranganatha** [comment]
+1

**Comment by MolotovCherry** [comment]
+1 This is annoying me a bit since I have an automatically managed branch that nobody needs to see any notification about.

**Comment by augustelalande** [comment]
+1 please

**Reply by ym-herby** [reply]
+100 this is monumental pain.

**Comment by fredrikodenelse** [comment]
+1

**Comment by J0E-E** [comment]
+1

**Comment by DerekMarcinyshyn** [comment]
+1

**Comment by bcavs** [comment]
+1

**Comment by blasdfaa** [comment]
+1

**Comment by kennerhartman** [comment]
+1

**Comment by diskria** [comment]
+1

**Comment by ThePat02** [comment]
+ 1

**Comment by kvfairchild** [comment]
+1

**Comment by vincentburkhardt** [comment]
+1

**Comment by kurylo** [comment]
+1

**Comment by Apachedrag427** [comment]
+1

**Comment by icecliffs** [comment]
+1

**Comment by IslamAbdelhakiim** [comment]
totally agree +1

**Comment by Griefed** [comment]
+1

Whilst I understand the potential reasoning for the existence of these banners, I have to agree that being able to disable them would be a gods-send. Especially when you have automated branches (no matter why), having said banners can become disruptive / annoying. Or even when working on a repo as a team, this can become quite loud.

**Comment by olafur-palsson** [comment]
We have a feature --> main --> production and we really don't want people merging production into main even though it's supposed to be an empty PR.

Then we have another project that' 
featureX --> main
featureY --> client-production (independently developed from main and customized project where main is a versioned library + template)

**Comment by nthomas-tc** [comment]
+1

**Comment by Oxyopiia** [comment]
absolute +1

**Comment by archae0pteryx** [comment]
please microsoft... preeeetty please?

**Comment by JeRoNZ** [comment]
It's really annoying. +1

**Comment by fredlemieux** [comment]
+1 perhaps add some settings in .github folder to disable this?

**Comment by Paethon** [comment]
Second that. Super annoying

**Comment by noatum-codyzornes** [comment]
+1 Would greatly appreciate this feature

**Comment by Creaous** [comment]
+1

**Comment by vbakkenes** [comment]
+1

**Comment by NonVideri** [comment]
+1

**Comment by ghost** [comment]
Does anyone even work at Github anymore ? Hello ? This is nearly 1.5 years old !

**Reply by NilsBaumgartner1994** [reply]
Seems like this wish:
- Is hard to encounter
- A security measurement which was decided
- other unknown challenges?

**Reply by ghost** [reply]
1. its not hard to encounter.
2. what has this to do with security at all ?
3. maybe.

**Reply by NilsBaumgartner1994** [reply]
2. Sorry i mean safety. Showing the user that there is a merge.

**Reply by olafur-palsson** [reply]
No matter what opinion you have on this programmers are just as dumb as the rest of humanity. People here are asking for the permission to not make it easy to shoot yourself in the foot.

**Comment by mariarebelkaps** [comment]
+1
This should be managed on the repository level, and at least allow administrator to disable the "compare & pull request" button on the banner

**Reply by NilsBaumgartner1994** [reply]
Maybe we should summarize the requirements ok this issue in order to support an easier implementation?

**Comment by zaidongy** [comment]
+1

**Comment by EsmeMaxwell** [comment]
+1

**Comment by hnsteyding** [comment]
+1

**Comment by danielgary** [comment]
I fully support this idea

**Comment by AdamCalculator** [comment]
+++

**Comment by satyapraneet63** [comment]
I agree. I don't want to merge my releasebranch into main. Main is ahead in development. :disappointed:

**Comment by msucevan** [comment]
There is still no way to do this? It's really annoying, I have a staging branch and I don't want to create a PR from staging

**Comment by gosuto-inzasheru** [comment]
+1

**Comment by chrisg-bytecodeio** [comment]
+1

**Comment by franc-in-action** [comment]
+1

**Comment by lukedabrzalski** [comment]
+1

**Comment by N-Fannilla** [comment]
+1, it is really annoying

**Comment by iamalhotra** [comment]
+1

**Comment by armouti** [comment]
+1

**Comment by bclark8923** [comment]
+1

**Comment by benbenz** [comment]
+1

**Comment by SabAlej** [comment]
+1

**Comment by MohamadAlAdraa** [comment]
+1

**Comment by Choi-JJunho** [comment]
+1

**Comment by sohale** [comment]
`++`

**Comment by tsktech** [comment]
+1

**Comment by escamoteur** [comment]
actually, it should be possible to disable this notification for any branch. If I have to make a release branch to fix an issue on an older release and I merge into that my changes so that it could get reviewed by the team, I now get this notification that my release branch could get merged into main which we don't want to do and I can't even dismiss that diablog

**Comment by SoCuul** [comment]
+1 you should be able to disable it per-branch or per-repo

**Comment by ChazUK** [comment]
Still can't believe this hasn't been implemented!

**Comment by martz-rgb** [comment]
+1

**Comment by deerbone** [comment]
+1

**Comment by BBrown4** [comment]
+1 come on, GitHub, at least say something 🙄

**Comment by koto-games** [comment]
+1

**Comment by SteveJamesDelta** [comment]
+1

**Comment by gmankab** [comment]
+1

**Comment by vovamishenskiy** [comment]
+1

**Comment by alifdarsim** [comment]
+1 Githubkun, notice me senpai

**Comment by symonxdd** [comment]
+69

**Reply by daniel-res** [reply]
> +69

almost 2 years later.... and no updates. 🏄

**Comment by SamGhAu** [comment]
+1

**Comment by Paul-blokparties** [comment]
+1

**Comment by tranphuquy19** [comment]
+1

**Comment by auser1337** [comment]
+1

**Comment by dantetemplar** [comment]
+1

**Comment by karol767** [comment]
agreed

**Comment by koto-games** [comment]
+2

**Comment by huertong** [comment]
+1

**Comment by LaCocoRoco** [comment]
+1

**Comment by rogerlin0330** [comment]
+1

**Comment by dhitchenor** [comment]
OMG, yes +1

**Comment by orielVeev** [comment]
+1

**Comment by PolyCatDev** [comment]
+1

**Comment by forderation** [comment]
+1

**Comment by EthanDelong** [comment]
+1

**Comment by Lafexlos** [comment]
another day another +1

**Comment by H4ppy-04** [comment]
I will also contribute my part of saying +1

**Comment by jhkrug** [comment]
+1

**Comment by IslamKolade** [comment]
+1

**Comment by Cynthia7979** [comment]
+1, can't believe this isn't a thing yet

**Comment by fruwe** [comment]
+1

**Comment by achyu-dev** [comment]
+1

**Comment by k0va1** [comment]
+1

**Comment by justoverclockl** [comment]
+1

**Comment by ChessDP** [comment]
+1 on this

**Comment by abtinmU** [comment]
+1
It's super annoying

**Comment by PatReis** [comment]
+1

**Comment by zuberspace** [comment]
+1

**Comment by overklassniy** [comment]
+1

**Comment by pascalecu** [comment]
+1

**Comment by refty10** [comment]
+1

**Comment by 1batu** [comment]
My suggestion is to add an option during branch protection to ensure that a specific branch is never selected as the source side of a Pull Request. This would be especially useful in scenarios where branches like develop are merged into master or main, but never merged back.

In these cases, setting the develop branch as the default and displaying it on the main page can be beneficial, while also providing an extra layer of protection to prevent new Pull Requests from accidentally targeting the master branch. Additionally, the banner suggesting that master needs to be merged back into develop should not appear in this scenario, making the workflow smoother.

**Comment by nusu** [comment]
+1

**Comment by eldoy** [comment]
+1

**Comment by Asteriuz** [comment]
+1

**Comment by setanarut** [comment]
+1 from 2024 September

**Comment by JonathanBout** [comment]
Why is this still not a thing after 2 years and 5 days 😭

**Comment by najibghadri** [comment]
Github wake up

**Comment by kzyqq00-Player** [comment]
+1

**Comment by taasmoe** [comment]
+1 :)

**Comment by Manuchexra** [comment]
Response #3

**Comment by jonaRumberg** [comment]
+1

**Comment by GenEugene** [comment]
+1

**Comment by lkellermann** [comment]
+1

**Comment by akkonrad** [comment]
it is annoying

**Comment by lesca** [comment]
+1

**Comment by Legnatbird** [comment]
+1

**Comment by voron5096** [comment]
+1

**Comment by pdm-pcb** [comment]
This has my upvote, too. As a solo maintainer... It's distracting clutter for me.

**Comment by flavius** [comment]
+1

Also consider orphan branches. It makes no sense to show them, even according to your own existing messages. Opening a PR from an orphan branch shows:

![image](https://github.com/user-attachments/assets/432ff6d9-1ed0-4c7f-95f2-9da1eb4bc550)

**Reply by dbaynard** [reply]
Related: [git-annex](https://git-annex.branchable.com/tips/centralized_git_repository_tutorial/on_GitHub/) repeatedly updates and pushes the `git-annex` branch. It would be great not to have a button that prompts undesirable behaviour.

**Comment by nuhkoca** [comment]
+1

**Comment by derDeno** [comment]
+1

**Comment by tomasalias** [comment]
+1

**Comment by josephcaxton** [comment]
+1

**Comment by ElectriPixie** [comment]
Can we at least get a dismiss button to make them go away? It is so infuriating having to look at these with no way to say NO!
So the only way I can find to make it go away is to create a pull request then close it without merging it, but that's just awful.

**Comment by tobias-farrenkopf** [comment]
+1

**Comment by Gladrat** [comment]
+1 for oprhan branches

**Comment by keetsta** [comment]
+1

**Comment by Seamus1989** [comment]
+1

**Comment by MadMud** [comment]
in my testing creating a single draft PR on that branch will prevent this banner being shown, even on further commits. Perhaps it will show up in a few days, though ;)

To be clear this is still annoying, but at least it clutters the PR section not the main view.

**Comment by MagneticNeedle** [comment]
+1

**Comment by br14n-sol** [comment]
+1

**Comment by labkey-willm** [comment]
+1

**Comment by labkey-jony** [comment]
+1

**Comment by KenKolano** [comment]
For the love of god. ENOUGH WITH THE +1 POSTS.

Use the Upvote button on the bottom of the description and if you want to follow the issue the "Subscribe" button above in the sidebar, spamming everyone that has subscribed is likely pointless. This issue is 3 years old now so there is likely to be nothing done about it.

**Reply by martindanielson** [reply]
+1

**Comment by hasirciogluhq** [comment]
+1

**Comment by EthanDelong** [comment]


**Reply by tizu69** [reply]
what?

**Comment by XVIIStarPlatinum** [comment]
+1, kinda surprised that this isn't a feature

**Comment by divakaivan** [comment]
+1

**Comment by shaneknapp** [comment]
+100

**Reply by jehlijos** [reply]
+1

**Comment by IsabelleLePivain** [comment]
Same use case here. We just switched to a [Gitflow workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow), using a `develop` branch like OP to merge features. Also, we are using `main` to trigger deploys to all of our environments.

We deploy code, hence update `main`, regularly - if not daily. 
As a consequence, this notification is very noisy.
Being able to deactivate it on a per branch basis would make using the Gitflow workflow much smoother.

**Comment by dennisvang** [comment]
A related problem: 

Merging a PR into a *non-default* branch does not close linked issues automatically.

Of course this makes sense, as there can only be one default branch, but it makes life more difficult when trying to maintain support branches.

From the [docs]:


>The special keywords in a pull request description are interpreted only when the pull request targets the repository's default branch. If the pull request targets any other branch, then these keywords are ignored, no links are created, and merging the PR has no effect on the issues.

[docs]: https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#about-linked-issues-and-pull-requests

**Comment by roncodes** [comment]
+1

**Comment by memray** [comment]
+111111

**Comment by shooqie** [comment]
+1, especially useful for personal repos where some branches are never meant to be merged into main

**Comment by martindanielson** [comment]
+1

**Comment by fennb** [comment]
+1

**Comment by J0E-E** [comment]
For those still here: I've made a TamperMonkey script to remove the banner. This could be further customized for conditionals, but I just never want to see it, so I removed it completely. 

https://github.com/J0E-E/gh-recently-touched-banner-remover

**Comment by Electron7-7** [comment]
+1
I hate feeling like I'm doing something stupid every time I "draft a new pull request" just so I can immediately close it and get rid of that banner.

**Comment by J0E-E** [comment]
I just noticed the link to my TM script wasn't working, I just updated it. It should work now.

**Comment by linkdata** [comment]
+1

**Comment by navjotkaur10** [comment]
+1

**Comment by lesca** [comment]
+1

**Comment by Frestein** [comment]
bump

**Comment by keenanjohnson** [comment]
Yes let's please implement this.

**Comment by jamiecurle** [comment]
Waiting patiently since 2017 for this.

I'd accept all AI being removed from the world in exchange for this one feature.

**Comment by ThermodynamicBeta** [comment]
+1, we have a pattern for automatic pushes that should be separate from other feature branches, would be great to never have the banners for that pattern

Edit: worked around this by pushing tags for the commits instead of branches.

**Comment by joczikszabi** [comment]
+1

**Comment by lachlansimpson** [comment]
+1

**Comment by aung-arata** [comment]
+1

**Comment by OMouta** [comment]
+1

**Comment by EthanDelong** [comment]
Nearly four years later and the developers of the site to host code for developers still are unable to implement a simple "dismiss" option at the bare minimum. That tracks.

**Comment by nikola-pandurski-mobisystems** [comment]
+1

**Comment by MisakaRehana** [comment]
+1

**Comment by MohamedAmjed** [comment]
wtf? Is this still not implemented?

**Reply by Reyzerbit** [reply]
thats what IM sayin

**Comment by cgranville** [comment]
+1

**Comment by reteps** [comment]
I saw https://github.com/orgs/community/discussions/35200#discussioncomment-13889786, go to

`uBlock Origin → Dashboard → My filters`

```css
github.com##div[class*="RecentlyTouchedBranches"]:has(a[href*="entire/checkpoints"])
```

and set the href to a include pattern to ignore. in my case, I used `entire/checkpoints`.

**Comment by Pocolip** [comment]
still want this ._.

**Comment by Zsupi** [comment]
+1

**Comment by engrrajonahmed** [comment]
+1

**Comment by MarlonTuckerAareon** [comment]
Has anyone in the github team even responded or acknowledged this feature request?

I find it hard to believe having distinct release branches is an uncommon git branching strategy. 

As another user pointed out, even a way to dismiss these would be nice. 

We need to give it an AI angle for anyone in Microsoft to move on it. How about they use copilot to analyse the codebase to work out if a PR is something the user might wish to do. Ignoring a simple git diff threshold would do that.

**Comment by RaphaelMaschinsen** [comment]
yes please

**Comment by rsoaresdev** [comment]
why is this not implemented yet...

**Comment by sebastianvarela** [comment]
yes please!

**Comment by domaine-stijn-slats** [comment]
+128904197845

**Comment by patricknelson** [comment]
Workaround for this if you use uBlock Origin. Just add a custom rule and swap out `BRANCH_NAME`:

```
github.com##div.flash:has(a[title="BRANCH_NAME"])
```

You can actually test this via the picker as well, just remove the `github.com` prefix and add it that way as well. 🙂

**Reply by yurtemre7** [reply]
I've fixed it with this:

```sh
div[data-component="Flash"]:has(a[href*="/tree/BRANCH_NAME"])
```

and uBlock Origin Lite does also work!

**Comment by dazzknowles** [comment]
+1

I'm moving all our repos to GitHub and bringing every branch along for the ride too so I'm getting a lot of these but they're artificial as some are for branches long superceded.

**Comment by crizmo** [comment]
+ 1
Much needed !

**Comment by jonathanpallant** [comment]
This banner is *so* annoying - I absolutely do not want to be encouraged to make a pull request between my release branches, because that would break everything.
